cmd/hypersdk-cli/transaction.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/spf13/cobra"
+
+	"github.com/ava-labs/hypersdk/abi/dynamic"
+	"github.com/ava-labs/hypersdk/api/jsonrpc"
+	"github.com/ava-labs/hypersdk/api/ws"
+	"github.com/ava-labs/hypersdk/auth"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+var txCmd = &cobra.Command{
+	Use:   "tx [action]",
+	Short: "Execute a transaction on the chain",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		// 1. Decode key
+		keyString, err := getConfigValue(cmd, "key")
+		if err != nil {
+			return fmt.Errorf("failed to get key from config: %w", err)
+		}
+		key, err := privateKeyFromString(keyString)
+		if err != nil {
+			return fmt.Errorf("failed to decode key: %w", err)
+		}
+
+		// 2. create client
+		endpoint, err := getConfigValue(cmd, "endpoint")
+		if err != nil {
+			return fmt.Errorf("failed to get endpoint: %w", err)
+		}
+		client := jsonrpc.NewJSONRPCClient(endpoint)
+
+		// 3. get abi
+		abi, err := client.GetABI(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get abi: %w", err)
+		}
+		// 4. get action name from args
+		if len(args) == 0 {
+			return errors.New("action name is required")
+		}
+		actionName := args[0]
+		_, found := abi.FindActionByName(actionName)
+		if !found {
+			return fmt.Errorf("failed to find action: %s", actionName)
+		}
+
+		typ, found := abi.FindTypeByName(actionName)
+		if !found {
+			return fmt.Errorf("failed to find type: %s", actionName)
+		}
+
+		// 5. create action using kvPairs
+		kvPairs, err := fillAction(cmd, typ)
+		if err != nil {
+			return err
+		}
+
+		jsonPayload, err := json.Marshal(kvPairs)
+		if err != nil {
+			return fmt.Errorf("failed to marshal kvPairs: %w", err)
+		}
+
+		actionBytes, err := dynamic.Marshal(abi, actionName, string(jsonPayload))
+		if err != nil {
+			return fmt.Errorf("failed to marshal action: %w", err)
+		}
+
+		_, _, chainID, err := client.Network(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get network info: %w", err)
+		}
+
+		base := &chain.Base{
+			ChainID:   chainID,
+			Timestamp: time.Now().Unix()*1000 + 60*1000,
+			MaxFee:    1_000_000,
+		}
```

Comment:
could we add a TODO in the code to handle the timestamp and max fee?

We'll need to use `EstimateUnits` to calculate the max fee and will need to know the maximum expiry time from the chain's rules to make sure the timestamp is valid (will require a new API call).